deeposlandia/aggregate_label.py

```diff
@@ -124,16 +59,18 @@ def aggregate_config(config, df):
     assert len(config['labels']) == df.shape[0], "should have the same size"
     result = {'folder_structure': config['folder_structure']}
     result['labels'] = []
-    for key, group in df.groupby(KEY_AGG):
-        label_id = int(group['id'].min())  # int conversion from int64 for JSON serialization
-        for _, label in group.iterrows():
-            if label['id'] == label_id:
-                d = config['labels'][label_id]
-                d['id'] = label_id
-                d['group_name'] = label[KEY_AGG]
-                d['aggregate'] = []
-            else:
-                d['aggregate'].append((label['id'], label['name']))
+    nb_labels = len(df.new_label.unique())
+    intgen = (i for i in range(nb_labels))
```

Comment:
I think you can replace it by `range()` which is already a generator in py3.

Reply:
Yum, it does not work. However, I can do `intgen = iter(range(nb_labels))` instead. I found this article to illustrate the point: http://treyhunner.com/2018/02/python-range-is-not-an-iterator/

Reply:
You're right. I forgot this point. Thanks for sharing the article.